lib/DDG/Spice/Timer.pm

```diff
@@ -13,18 +13,50 @@ code_url 'https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/lib/DDG/
 attribution twitter => 'mattr555',
             github => ['https://github.com/mattr555/', 'Matt Ramina'];
 
-triggers startend => ['timer', 'countdown', 'alarm'];
-triggers start => ['time', 'timer for'];
+my @triggers = qw(timer countdown alarm);
+triggers startend => @triggers;
 
 spice call_type => 'self';
 
 handle remainder => sub {
-    return if lc($req->query_raw) =~ /^time($|[^r ][\S]+$)/;  #filter out queries with like "time" or "Time::Piece"
-    return if lc($req->query_raw) =~ /^(timer|countdown|alarm)\S+$/; #filter timer.x or countdown.x or alarm.x
-    return unless /^( ?([\d.]+ ?(m(in((ute)?s?)?)?|s(ec((ond)?s?)?)?|h(ours?)?|hr)|online) ?)+$/ ||
-        $_ eq '' ||
-        /^( ?((\d{1,2}:)?\d{1,2}:\d{2}) ?)/;
-    return '';
+    my $qry = $_;
+    my $raw = lc($req->query_raw);
+    my $trgx = join('|', @triggers);
+
+    # check to make sure the query matches a trigger perfectly
+    if($qry eq '' && in_array($raw, @triggers)) {
```

Comment:
@davidofferman awesome! I was thinking that maybe instead of implementing an `in_array` subroutine you could use something like this? Tell me what you think! http://stackoverflow.com/a/6475315